elementary/monitor/data_monitoring/schema.py

```diff
@@ -28,23 +28,48 @@ class ResourceType(Enum):
     SOURCE_FRESHNESS = "source_freshness"
 
 
-class SelectorFilterSchema(BaseModel):
+class SupportedFilterTypes(Enum):
+    IS = "is"
+
+
+class FilterSchema(BaseModel):
+    # The relation between values is OR.
+    values: List[Any]
+    type: SupportedFilterTypes = SupportedFilterTypes.IS
+
+    class Config:
+        # Make sure that serializing Enum return values
+        use_enum_values = True
+
+
+class StatusFilterSchema(FilterSchema):
+    values: List[Status]
+
+
+class ResourceTypeFilterSchema(FilterSchema):
+    values: List[ResourceType]
+
+
+class FiltersSchema(BaseModel):
     selector: Optional[str] = None
     invocation_id: Optional[str] = None
-    invocation_time: Optional[str] = None
+    invocation_time: Optional[str] = Field(default=None)
```

Comment:
Because the default field is not mutable so you can just use None 🙂 
The Field default factory is for dynamic defaults and mutable defaults like list and dict
```suggestion
    invocation_time: Optional[str] = None
```